utils/S3UtilsMongoClient.js

```diff
@@ -0,0 +1,246 @@
+            }
+        }
+        entry.value.replicationInfo.backends.forEach(rep => {
+            // count it in the replication destination location if replication compeleted
+            if (rep.status === 'COMPLETED') {
+                if (results.location[rep.site]) {
+                    results.location[rep.site] += size;
+                } else {
+                    results.location[rep.site] = size;
+                }
+            }
+        });
+        return {
+            data: results,
+            error: null,
+        };
+    }
+
+    _handleResults(res, isVersioned) {
+        let totalNonCurrentCount = 0;
+        let totalCurrentCount = 0;
+        const totalBytes = { curr: 0, prev: 0 };
+        const locationBytes = {};
+        const dataMetrics = {
+            bucket: {},
+            location: {},
+            account: {},
+        };
+
+        Object.keys(res).forEach(metricLevel => {
+            // metricLevel can only be 'bucket', 'location' or 'account'
+            if (validStorageMetricLevels.has(metricLevel)) {
+                Object.keys(res[metricLevel]).forEach(resource => {
+                    // resource can be the name of bucket, location or account
+                    const resourceName = metricLevel === 'location' ? this._getLocName(resource) : resource;
+                    if (!dataMetrics[metricLevel][resourceName]) {
+                        dataMetrics[metricLevel][resourceName] = {
+                            usedCapacity: {
+                                current: 0,
+                                nonCurrent: 0,
+                            },
+                            objectCount: {
+                                current: 0,
+                                nonCurrent: 0,
+                                deleteMarker: 0,
+                            },
+                        };
+                    }
+                    const {
+                        masterCount,
+                        masterData,
+                        nullCount,
+                        nullData,
+                        versionCount,
+                        versionData,
+                        deleteMarkerCount,
+                    } = res[metricLevel][resourceName];
+
+                    dataMetrics[metricLevel][resourceName].usedCapacity.current += nullData + masterData;
+                    dataMetrics[metricLevel][resourceName].objectCount.current += nullCount + masterCount;
+
+                    if (isVersioned) {
+                        dataMetrics[metricLevel][resourceName].usedCapacity.nonCurrent += versionData - masterData;
+                        dataMetrics[metricLevel][resourceName].usedCapacity.nonCurrent = Math.max(dataMetrics[metricLevel][resourceName].usedCapacity.nonCurrent, 0);
+                        dataMetrics[metricLevel][resourceName].objectCount.nonCurrent
+                            += versionCount - masterCount - deleteMarkerCount;
```

Comment:
Why do we need to subtract deleteMarkers from versions here? deleteMarker(s) are not considered as versions?

Reply:
we have 3 metrics for `usedCapacity`: current, nonCurrent, deleteMarker
I feel like it's more logical to have them separated, we can add them up when needed